qa/rpc-tests/test_framework/test_framework.py

```diff
@@ -100,8 +100,12 @@ def main(self):
         parser.add_option("--tracerpc", dest="trace_rpc", default=False, action="store_true",
                           help="Print out all RPC calls as they are made")
         self.add_options(parser)
-        (self.options, self.args) = parser.parse_args()
+        opts, args = parser.parse_args()
+        return (opts, args)
 
+    def main(self):
+        if not (hasattr(self, "options") or hasattr(self, "args")):
```

Comment:
I think it makes mores sense for this "or" to be an "and".   The presence of both indicates that `self.parse_options_args()` has run, the absence of either indicates, either that it hasn't run, or that some subsequent operation has changed the outcome.   Basically the if statement should mean:

   If the thing (`parse_options_args`) has produced the normal expected output, then run it. 

 Soo...   maybe an "and" would be better.